astropy_helpers/commands/build_sphinx.py

```diff
@@ -170,18 +172,22 @@ def run(self):
         # prevents a continuous updating at the terminal, but there's no
         # apparent way around this.
         if self.warnings_returncode:
-            proc = subprocess.Popen([sys.executable],
+            proc = subprocess.Popen([sys.executable, '-c', subproccode.encode('utf-8')],
                                     stdin=subprocess.PIPE,
                                     stdout=subprocess.PIPE,
                                     stderr=subprocess.STDOUT)
-            stdo, _ = proc.communicate(subproccode.encode('utf-8'))
-            stdo = stdo.decode('utf-8')
 
-            print(stdo)
+            stdolines = []
+            with proc.stdout:
+                for line in iter(proc.stdout.readline, b''):
+                    line = line.strip(b'\n')
+                    stdolines.append(line)
+                    print(line.decode('utf-8'))
 
-            stdolines = stdo.splitlines()
+            # Poll to set proc.retcode
+            proc.poll()
```

Comment:
It might be safer to guard this with something like:
```
while proc.poll() is None:
    time.sleep(.5)
```
I've seen travis (and maybe RTD) sit for a while at the end of the process and it's not entirely clear why, so it's conceivable that something weird is going on that might lead to `stdout` getting closed before the process has actually finished.